server/async_functions.py

```diff
@@ -0,0 +1,27 @@
+"""
+Some helper functions for common async tasks.
+"""
+import asyncio
+from typing import Callable, List, Optional
+
+
+async def gather_without_exceptions(
```

Comment:
If we want to handle exceptions in the caller, then we can use `asyncio.as_completed` like this:

```
import asyncio


async def fail(t):
    await asyncio.sleep(t)
    print(f"Failing {t}")
    raise Exception


async def do():
    jobs = [fail(i) for i in range(10)]
    for f in asyncio.as_completed(jobs):
        try:
            await f
        except Exception:
            print("Caught an exception!")


asyncio.get_event_loop().run_until_complete(do())
```

Otherwise, we could just wrap our tasks with exception catching handlers instead of passing in callbacks.

Reply:
Ok nice. That's much better than a callback.